sql/src/main/java/io/crate/execution/dsl/package-info.java

```diff
@@ -0,0 +1,107 @@
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+/**
+ * <p>
+ *  This package contains classes for the execution domain-specific language (DSL).
+ *  This DSL is used to describe how queries are executed.
+ *  <br />
+ *
+ *  "execution" here refers to the *generic* execution framework used to
+ *  execute DQL or DML statements. (DML restricted to those involving queries - like
+ *  insert from subquery, or update by query).
+ *  <br />
+ *
+ *  Some DML statements and DDL statements may be executed in a specialized form.
+ *  Their execution isn't described using the DSL contained in this package.
+ * </p>
+ *
+ * <h2>Phases</h2>
+ *
+ * <p>
+ *  A ExecutionPhase is the "core" of the execution DSL. The interfaces are:
+ * </p>
+ * <ul>
+ *  <li>{@link io.crate.execution.dsl.phases.ExecutionPhase}</li>
+ *  <li>{@link io.crate.execution.dsl.phases.UpstreamPhase}</li>
+ * </ul>
+ * <p>
+ *  There are three categories of ExecutionPhases:
+ * </p>
+ *
+ * <ul>
+ *  <li>
+ *      Phases that describe *how* to produce *what* data<br />
+ *      ({@link io.crate.execution.dsl.phases.CollectPhase})
+ *  </li>
+ *  <li>
+ *      Phases that describe how to receive and process data
+ *      ({@link io.crate.execution.dsl.phases.MergePhase}, {@link io.crate.execution.dsl.phases.NestedLoopPhase})
+ *  </li>
+ *  <li>
+ *      Phases that instruct the executor to create some kind of "context" which may be used by other phases.
+ *      ({@link io.crate.execution.dsl.phases.FetchPhase})
+ *  </li>
+ * </ul>
+ *
+ * <p>
+ *  Data generally "flows through" phases. For example:
+ *  (Context phases are an exception here)
+ * </p>
+ * <pre>
+ *     select * from t1
+ *
+ *     Would be represented as
+ *
+ *     CollectPhase (Running on [node1, node2, node3])
+ *          |
+ *     MergePhase (Running on [node1])
+ *
+ *     The data flows from CollectPhase to MergePhase
+ * </pre>
+ * <p>
+ *     This data flow is implicitly defined in a concrete {@link io.crate.planner.ExecutionPlan} (e.g. {@link io.crate.planner.Merge})
+ *     due to the order how the phases appear in the plan.
+ *     <br />
+ *     Due to that the execution layer uses a {@link io.crate.execution.dsl.phases.NodeOperation}.
+ *     That is a wrapped/annotated ExecutionPhase which contains explicit
```

Comment:
Would skip the words `due to that` because I don't see a connection to the previous sentence. It just happens to be the runtime presentation.